__testUtils/mockAddressBook.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  addStringNoLocale,
+  addUrl,
+  mockSolidDatasetFrom,
+  mockThingFrom,
+  setThing,
+} from "@inrupt/solid-client";
+import { acl, dc, foaf, rdf, vcard } from "rdf-namespaces";
+import { chain } from "../src/solidClientHelpers/utils";
+import { aliceWebIdUrl } from "./mockPersonResource";
+import { vcardExtras } from "../src/addressBook";
+import { getContactIndexDefaultUrl } from "../src/models/contact";
+import { getAddressBookIndexDefaultUrl } from "../src/models/addressBook";
+
+const defaultContainerIri = "https://user.example.com/contacts";
+
+export function mockAddressBookThing(options = {}) {
+  const containerIri = options.containerIri || defaultContainerIri;
+  const owner = options.owner || aliceWebIdUrl;
+  const indexUrl =
+    options.indexUrl || getAddressBookIndexDefaultUrl(containerIri);
+  const groupsUrl =
+    options.groupsUrl ?? getContactIndexDefaultUrl(containerIri, vcard.Group);
+  const peopleUrl =
+    options.peopleUrl ?? getContactIndexDefaultUrl(containerIri, foaf.Person);
+  const mainIndexIri = `${indexUrl}#this`;
+  debugger;
```

Comment:
Remove debugger - surprised the linter allows this. Also, groups and people should be able to be removed from the mock.

Reply:
I'll see what I can do about the mocking methods - they're a bit convoluted right now, so yeah, could probably clear up a bit there.